benchmark/ext_helpers/kmeans_dpcpp.py

```diff
@@ -0,0 +1,48 @@
+import dpctl.tensor as dpt
+
+from sklearn.exceptions import NotSupportedByEngineError
+
+from sklearn_numba_dpex.kmeans.engine import KMeansEngine
+
+import kmeans_dpcpp as kdp
+
+
+class KMeansDPCPPEngine(KMeansEngine):
+    def init_centroids(self, X):
+        init = self.init
+        if isinstance(init, str) and init == "k-means++":
+            raise NotSupportedByEngineError(
+                "kmeans_dpcpp does not support k-means++ init."
+            )
+        return super().init_centroids(X)
+
+    def kmeans_single(self, X, sample_weight, centers_init_t):
+        n_samples, n_features = X.shape
+        device = X.device.sycl_device
+
+        assignments_idx = dpt.empty(n_samples, dtype=dpt.int32, device=device)
+        res_centroids_t = dpt.empty_like(centers_init_t)
```

Comment:
Should the device for `res_centroids_t` be specified?

Reply:
`empty_like` will create `res_centroids_t` on the same device than `centers_init_t`, which should be on the appropriate device already, so it's not necessary.

FYI device mismatchs are easy to spot because kernels won't accept inputs that are on different devices (raise errors instead) so if the data allocation is done on wrong devices it will break downstream.